### PR TITLE
Fix #1845: Replace Language code with icon in AudioBar

### LINKED ISSUE #1845: Replace Language code with icon in AudioBar
Currently Audio Bar in exploration has a language code, as shown below:
Screenshot_1600232780

We should replace the language code to [icon](https://drive.google.com/file/d/1jPRimZBxcPHeEHwlmTUcLrDcwZyVt0gl/view?usp=sharing) as shown below:
Screenshot 2020-09-16 at 10 37 42 AM

Reference: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/d74f0571-471f-4ea6-a3a5-32e135253c56/

The tests in AudioFragmentTest need to be updated (see the TODOs there).

### PR DESCRIPTION
**Pull request by MaskedCarrot**
## Explanation
Fixes #1845

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### Tests

 class | Espresso | Roboelectric
--- | --- | --- 
ExplorationActivityTest | ExplorationActivityTestExpresso | ExplorationActivityTestRobo
AudioFragmentTest | AudioFragmentTestEspresso | AudioFragmentTestRobo

### CODE REVIEW

**Comment by MaskedCarrot** [comment]
All tests are passing in ExplorationActivityTest but only roboelectric tests are passing in AudioFragmentTest.

File: app/src/main/res/layout/audio_fragment.xml
Comment (anandwana001) [inline app/src/main/res/layout/audio_fragment.xml:65]: apply reformat and rearrange code from the menu in Android Studio.

File: app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt]: Is this test not working?
If this test is correct and good to go, please remove the ignore annotation.

File: app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt:549]: please add a screenshot showing these tests are passing on Espresso and Robolectric.

File: app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt:141]: Run these 3 tests individually on espresso and see if it passes or not. 
Also, add 2 screenshots running all the test cases on espresso and robolectric showing the test result.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @MaskedCarrot 
Added a few updates. PTAL

**Comment by rt4914** [comment]
Will review this later as @anandwana001 has already suggested changes.

File: app/src/main/res/layout/audio_fragment.xml
Comment (MaskedCarrot) [reply app/src/main/res/layout/audio_fragment.xml:65]: done

File: app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt
Comment (MaskedCarrot) [reply app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt]: This test is not working. But I think we should have this test. What do you suggest?

File: app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt
Comment (MaskedCarrot) [reply app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt:549]: added the screenshots in pr description but Espresso tests are not passing on AudioFragmentTest

File: app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt
Comment (MaskedCarrot) [reply app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt:141]: Two of the three tests associated with `ivAudioLang` are passing. I have removed the third test case `testAudioFragment_openFragment_showsDefaultAudioLanguageAsHindi` as it is failing on both roboelectric and espresso

Comment (MaskedCarrot) [reply app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt:141]: Apart from the two tests assosiated with `ivAudioLang` and the test `testAudioFragment_openFragment_showsFragment()` all other tests are failing for `audioFragmentTest` on Espresso

File: app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt]: What causing failure of this test?

File: app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt:141]: Can we fix this test?

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_audio_lang_24px.xml:12]: Press `Control + Alt + L` - reformatting shortcut to fix the formatting.

```
<vector xmlns:android="http://schemas.android.com/apk/res/android"
  android:width="24dp"
  android:height="24dp"
  android:viewportWidth="24"
  android:viewportHeight="24">
  <path
    android:fillColor="#fff"
    android:pathData="M21.04,5.93C19.04,1.92 15.45,0.21 11.38,0.05A12.21,12.21 0,0 0,5.04 1.63,9.77 9.77,0 0,0 0.04,9.45a8.84,8.84 0,0 0,2 6.44,1.75 1.75,0 0,1 0.3,2.06c-0.47,0.89 -1,1.74 -1.56,2.59 -0.21,0.33 -0.41,0.64 -0.21,1a0.9,0.9 0,0 0,1 0.51,12.16 12.16,0 0,0 2.53,-0.7c1,-0.44 1.87,-1.09 2.81,-1.62a1.06,1.06 0,0 1,0.72 -0.09,11.52 11.52,0 0,0 9.43,-1.17A9.5,9.5 0,0 0,21.04 5.93ZM9.25,12.05a0.62,0.62 0,0 1,-0.51 -0.25,6.73 6.73,0 0,1 -0.43,-1.19c-0.11,-0.39 -0.29,-0.6 -0.74,-0.54a3.43,3.43 0,0 1,-1.06 0c-0.59,-0.1 -0.83,0.19 -1,0.7 -0.07,0.29 -0.2,0.56 -0.28,0.85s-0.29,0.56 -0.65,0.43 -0.34,-0.44 -0.23,-0.76c0.65,-1.87 1.29,-3.74 2,-5.6 0.12,-0.33 0.1,-0.83 0.59,-0.84s0.53,0.5 0.65,0.84c0.64,1.84 1.28,3.67 1.91,5.51 0.05,0.13 0.08,0.27 0.16,0.5C9.53,11.79 9.4,12.05 9.25,12.05ZM17.5,10.6a0.54,0.54 0,0 0,-0.4 0.23,7.94 7.94,0 0,1 -1.86,3.08l1.6,1c0.28,0.16 0.57,0.32 0.38,0.69s-0.51,0.33 -0.82,0.15c-0.62,-0.37 -1.23,-0.76 -1.8,-1.11l-2.31,1.17c-0.28,0.14 -0.61,0.2 -0.77,-0.18s0.1,-0.55 0.43,-0.7c0.65,-0.29 1.29,-0.63 1.92,-0.94l-1.2,-1.57c-0.2,-0.28 -0.26,-0.59 0.07,-0.78s0.54,0 0.74,0.28c0.32,0.47 0.68,0.9 1.06,1.4a6.39,6.39 0,0 0,1.66 -2.7L12.04,10.62c-0.32,0 -0.56,-0.11 -0.56,-0.48s0.24,-0.48 0.57,-0.48h2.32a3.16,3.16 0,0 1,0 -0.88c0.06,-0.21 0.27,-0.5 0.45,-0.53s0.49,0.2 0.49,0.54 0,0.53 0,0.87L17.44,9.66c0.36,0 0.77,0 0.78,0.48S17.86,10.56 17.5,10.58Z" />
  <path
    android:fillColor="#fff"
    android:pathData="M6.14,9.11L7.79,9.11L6.96,6.73C6.65,7.61 6.4,8.35 6.14,9.11Z" />
</vector>

```

File: app/src/main/res/layout/audio_fragment.xml
Comment (rt4914) [inline app/src/main/res/layout/audio_fragment.xml:74]: Remove `currentLanguageCode` from ViewModel and its related code if not needed or make it private.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@MaskedCarrot Suggested nit changes.

File: app/src/main/res/layout/audio_fragment.xml
Comment (MaskedCarrot) [reply app/src/main/res/layout/audio_fragment.xml:74]: @rt4914 I  made the `viewModel.currentLanguageCode` as content description and then i check the content description of the image in test in audioFragmentTest

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (MaskedCarrot) [reply app/src/main/res/drawable/ic_audio_lang_24px.xml:12]: done

File: app/src/main/res/layout/audio_fragment.xml
Comment (rt4914) [reply app/src/main/res/layout/audio_fragment.xml:74]: Yeah, I forgot to remove this comment when I saw the tests. Thanks.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks. Will wait for @anandwana001 review.

**Comment by anandwana001** [comment]
@MaskedCarrot could you comment on my comments. Need before further review.

File: app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt
Comment (MaskedCarrot) [reply app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt]: When i run this test
```
@Test
  fun testAudioFragment_openFragment_showsDefaultAudioLanguageAsHindi() {
    addMediaInfo()
    launch<AudioFragmentTestActivity>(
      createAudioFragmentTestIntent(
        PROFILE_ID_DEFAULT_AUDIO_LANGUAGE_HINDI
      )
    ).use {
      testCoroutineDispatchers.runCurrent()
      onView(withId(R.id.ivAudioLanguage)).check(matches(withContentDescription("hi")))
    }
  }
```
then the content description remains as `en` and does not change to `hi`

Comment (MaskedCarrot) [reply app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt]: This test is also failing on espresso.

Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt]: We need to fix this failure as it looks like the implementation fail on production app.

Comment (MaskedCarrot) [reply app/src/sharedTest/java/org/oppia/android/app/player/audio/AudioFragmentTest.kt]: @anandwana001 I tried to figure out what the problem and found this `ProfileId 0 does not match an existing Profile`

**Comment by anandwana001** [comment]
@ShridharGoel Could you help here with the tests?

**Comment by ShridharGoel** [comment]
The Espresso tests seem to be failing because of use of Data Binding while setting the content description.

**Comment by MaskedCarrot** [comment]
I looked a little deeper into the failing test case and found that:   
A new profile is created in `AudioFragmentPresenter` with the line `profileId = ProfileId.newBuilder().setInternalId(internalProfileId).build()`
but the function `getProfile` of `ProfileManagementController` fails to retrive the newly created profile
```
  fun getProfile(profileId: ProfileId): DataProvider<Profile> {
    return profileDataStore.transformAsync(GET_PROFILE_PROVIDER_ID) {
      val profile = it.profilesMap[profileId.internalId]
      if (profile != null) {
        AsyncResult.success(profile)
      } else {
        AsyncResult.failed(
          ProfileNotFoundException(
            "ProfileId ${profileId.internalId} does" +
              " not match an existing Profile"
          )
        )
      }
    }
  }
``` 
ths function returns an empty  map  ` it.profilesMap[profileId.internalId]` hence a new profile is created everytime with invalid audio language. It will be really helpfull to get a few pointers on how to solve this issue.

**Comment by Luffy18346** [comment]
@rt4914 PTAL

**Comment by rt4914** [comment]
> I looked a little deeper into the failing test case and found that:
> A new profile is created in `AudioFragmentPresenter` with the line `profileId = ProfileId.newBuilder().setInternalId(internalProfileId).build()`
> but the function `getProfile` of `ProfileManagementController` fails to retrive the newly created profile
> 
> ```
>   fun getProfile(profileId: ProfileId): DataProvider<Profile> {
>     return profileDataStore.transformAsync(GET_PROFILE_PROVIDER_ID) {
>       val profile = it.profilesMap[profileId.internalId]
>       if (profile != null) {
>         AsyncResult.success(profile)
>       } else {
>         AsyncResult.failed(
>           ProfileNotFoundException(
>             "ProfileId ${profileId.internalId} does" +
>               " not match an existing Profile"
>           )
>         )
>       }
>     }
>   }
> ```
> 
> ths function returns an empty map ` it.profilesMap[profileId.internalId]` hence a new profile is created everytime with invalid audio language. It will be really helpfull to get a few pointers on how to solve this issue.

@MaskedCarrot 
`ProfileId.newBuilder().setInternalId(internalProfileId).build()` 
does not create a profile. It does converts a profileid of integer type to a profile id of "ProfileId" type. Its just of type conversion.

**Comment by anandwana001** [comment]
@MaskedCarrot This pr is correcting all the invalid things. Thanks for mentioning it. 
https://github.com/oppia/oppia-android/pull/2425

**Comment by MaskedCarrot** [comment]
@anandwana001  thanks, I will update this pr once #2425 gets merged.

**Comment by MaskedCarrot** [comment]
@anandwana001 the test `testAudioFragment_withHindiAudioLanguageProfile_showsHindiAudioLanguage()` is incorrectly passing on roboelectric as it checked the wrong text in #2425. Changing the text to `hi` makes it pass on espresso but it fails on roboelectric.

**Comment by anandwana001** [comment]
As we are removing `tvAudioLanguage` in this PR, that test needs to be removed in this PR only. Also, please resolve the conflict.

**Comment by MaskedCarrot** [comment]
closing this PR because I was not able to push a commit without force push.

**Pull request closed without merging**
